modules/calib3d/src/rhorefc.cpp

```diff
@@ -0,0 +1,2426 @@
+                         {x2Y2-x1Y1,y2Y2-y1Y1,(Y1-Y2)},
+                         {-x2Y2    ,-y2Y2    ,(Y2   )},
+                         {x2Y2-x3Y3,y2Y2-y3Y3,(Y3-Y2)}};*/
+    float minor[2][4] = {{x0-x2,x1-x2,x2   ,x3-x2},
+                         {y0-y2,y1-y2,y2   ,y3-y2}};
+    float major[3][8] = {{x2X2-x0X0,x2X2-x1X1,-x2X2    ,x2X2-x3X3,x2Y2-x0Y0,x2Y2-x1Y1,-x2Y2    ,x2Y2-x3Y3},
+                         {y2X2-y0X0,y2X2-y1X1,-y2X2    ,y2X2-y3X3,y2Y2-y0Y0,y2Y2-y1Y1,-y2Y2    ,y2Y2-y3Y3},
+                         { (X0-X2) , (X1-X2) , (X2   ) , (X3-X2) , (Y0-Y2) , (Y1-Y2) , (Y2   ) , (Y3-Y2) }};
+
+    /**
+     * int i;
+     * for(i=0;i<8;i++) major[2][i]=-major[2][i];
+     * Eliminate column 0 of rows 1 and 3
+     * R(1)=(x0-x2)*R(1)-(x1-x2)*R(0),     y1'=(y1-y2)(x0-x2)-(x1-x2)(y0-y2)
+     * R(3)=(x0-x2)*R(3)-(x3-x2)*R(0),     y3'=(y3-y2)(x0-x2)-(x3-x2)(y0-y2)
+     */
+
+    float scalar1=minor[0][0], scalar2=minor[0][1];
+    minor[1][1]=minor[1][1]*scalar1-minor[1][0]*scalar2;
+
+    major[0][1]=major[0][1]*scalar1-major[0][0]*scalar2;
+    major[1][1]=major[1][1]*scalar1-major[1][0]*scalar2;
+    major[2][1]=major[2][1]*scalar1-major[2][0]*scalar2;
+
+    major[0][5]=major[0][5]*scalar1-major[0][4]*scalar2;
+    major[1][5]=major[1][5]*scalar1-major[1][4]*scalar2;
+    major[2][5]=major[2][5]*scalar1-major[2][4]*scalar2;
+
+    scalar2=minor[0][3];
+    minor[1][3]=minor[1][3]*scalar1-minor[1][0]*scalar2;
+
+    major[0][3]=major[0][3]*scalar1-major[0][0]*scalar2;
+    major[1][3]=major[1][3]*scalar1-major[1][0]*scalar2;
+    major[2][3]=major[2][3]*scalar1-major[2][0]*scalar2;
+
+    major[0][7]=major[0][7]*scalar1-major[0][4]*scalar2;
+    major[1][7]=major[1][7]*scalar1-major[1][4]*scalar2;
+    major[2][7]=major[2][7]*scalar1-major[2][4]*scalar2;
+
+    /**
+     * Eliminate column 1 of rows 0 and 3
+     * R(3)=y1'*R(3)-y3'*R(1)
+     * R(0)=y1'*R(0)-(y0-y2)*R(1)
+     */
+
+    scalar1=minor[1][1];scalar2=minor[1][3];
+    major[0][3]=major[0][3]*scalar1-major[0][1]*scalar2;
+    major[1][3]=major[1][3]*scalar1-major[1][1]*scalar2;
+    major[2][3]=major[2][3]*scalar1-major[2][1]*scalar2;
+
+    major[0][7]=major[0][7]*scalar1-major[0][5]*scalar2;
+    major[1][7]=major[1][7]*scalar1-major[1][5]*scalar2;
+    major[2][7]=major[2][7]*scalar1-major[2][5]*scalar2;
+
+    scalar2=minor[1][0];
+    minor[0][0]=minor[0][0]*scalar1-minor[0][1]*scalar2;
+
+    major[0][0]=major[0][0]*scalar1-major[0][1]*scalar2;
+    major[1][0]=major[1][0]*scalar1-major[1][1]*scalar2;
+    major[2][0]=major[2][0]*scalar1-major[2][1]*scalar2;
+
+    major[0][4]=major[0][4]*scalar1-major[0][5]*scalar2;
+    major[1][4]=major[1][4]*scalar1-major[1][5]*scalar2;
+    major[2][4]=major[2][4]*scalar1-major[2][5]*scalar2;
+
+    /**
+     * Eliminate columns 0 and 1 of row 2
+     * R(0)/=x0'
+     * R(1)/=y1'
+     * R(2)-= (x2*R(0) + y2*R(1))
+     */
+
+    scalar1=minor[0][0];
+    major[0][0]/=scalar1;
```

Comment:
I'm not sure if compiler can optimize out the repetitive divisions by the same number. It's better to precompute 1.f/scalar1 and then multiply the results by it. Division is an order of magnitude slower operation than multiplication.